editor/plugins/texture_editor_plugin.cpp

```diff
@@ -61,11 +70,27 @@ void TexturePreview::_notification(int p_what) {
 				metadata_label->add_theme_font_override(SceneStringName(font), metadata_label_font);
 			}
 
+			bg_rect->set_color(get_theme_color(SNAME("dark_color_2"), EditorStringName(Editor)));
 			checkerboard->set_texture(get_editor_theme_icon(SNAME("Checkerboard")));
 		} break;
 	}
 }
 
+void TexturePreview::_draw_outline() {
+	if (outline_width == 0) {
+		return;
+	}
+	const Rect2 outline_rect = Rect2(Vector2(), texture_display->get_size()).grow((float)outline_width / 2);
+	const Color outline_border_color = get_theme_color(SNAME("extra_border_color_1"), EditorStringName(Editor));
```

Comment:
Shouldn't this be cached (on `NOTIFICATION_THEME_CHANGED`)? :thinking:

Reply:
Not sure about it. After `NOTIFICATION_THEME_CHANGED` function `TexturePreview::_draw_outline()` will be also called.
Also there are some examples where it is done the same way: 
- https://github.com/godotengine/godot/blob/1aaf20b1f1f3ab59f04db74b166540327a3f1f90/editor/animation_track_editor.cpp#L5936-L5940
- https://github.com/godotengine/godot/blob/1aaf20b1f1f3ab59f04db74b166540327a3f1f90/editor/animation_bezier_editor.cpp#L691-L707

Reply:
> After `NOTIFICATION_THEME_CHANGED` function `TexturePreview::_draw_outline()` will be also called.

Yeah, this I'm aware of. I was rather considering that `TexturePreview::_draw_outline()` can be called more often, not only on theme change, so always refetching the border color from the theme seems superfluous. But this is a minor issue, should be fine to leave as is (especially given it's already being done like that across the codebase).

Reply:
Theme changes are always accompanied by `NOTIFICATION_THEME_CHANGED` and every value can be cached. There is still some code that doesn't use cache, because it's more convenient this way, or the code is just old.

Reply:
Caching added.